### PR TITLE
Use GetRideTypeDescriptor() in more places

### PR DESCRIPTION
**Pull request by Gymnasiast**
Also guarantee that there will always be an RTD to return, preventing crashes from hacked rides or ride types only present in newer versions of OpenRCT2.

### CODE REVIEW

File: src/openrct2-ui/windows/Ride.cpp [hunk 1]
```diff
@@ -2536,7 +2535,7 @@ static rct_string_id window_ride_get_status_vehicle(rct_window* w, Formatter& ft
         return STR_EMPTY;
 
     auto stringId = VehicleStatusNames[static_cast<size_t>(vehicle->status)];
-    if ((RideTypeDescriptors[ride->type].Flags & RIDE_TYPE_FLAG_SINGLE_SESSION)
+    if ((ride->GetRideTypeDescriptor().Flags & RIDE_TYPE_FLAG_SINGLE_SESSION)
```
Comment (IntelOrca) [inline src/openrct2-ui/windows/Ride.cpp:2538]:
```suggestion
    if ((ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_SINGLE_SESSION))
```

**Comment by IntelOrca** [comment]
There over 95 occurances of `ride->GetRideTypeDescriptor().HasFlag`. Do you think we should have a helper on Ride?
`ride->HasRideTypeDescriptorFlag`?

**Comment by Gymnasiast** [comment]
I'm not entirely sure, I expect many functions to use the RTD more than once. In cases like those, doing
```C++
const auto& rtd = ride->GetRideTypeDescriptor();
if (rtd.HasFlag(RIDE_TYPE_FLAG_FOO)) 
{
    // etc.
}
```
would make more sense.

**Comment by IntelOrca** [comment]
@Gymnasiast certainly in those cases. I was talking about for any case where you had a single method with only a call to check a flag. I don't know many of those there are.

**Comment by Gymnasiast** [comment]
I'll look into tomorrow, then.

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -397,7 +397,7 @@ money32 Ride::CalculateIncomePerHour() const
         priceMinusCost -= GetShopItemDescriptor(currentShopItem).Cost;
     }
 
-    currentShopItem = (lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO) ? RideTypeDescriptors[type].PhotoItem
+    currentShopItem = (lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO) ? ::GetRideTypeDescriptor(type).PhotoItem
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp:400]: You can just use the ride version here. Type is the member variable.

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -7333,10 +7320,7 @@ uint64_t Ride::GetAvailableModes() const
 
 const RideTypeDescriptor& Ride::GetRideTypeDescriptor() const
 {
-    if (type >= std::size(RideTypeDescriptors))
-        return DummyRTD;
-
-    return RideTypeDescriptors[type];
+    return ::GetRideTypeDescriptor(type);
 }
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp:7324]: This I the only function in this file that needs to use `::` the ride members can call this version and not need to pass the type

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**